File: client/src/app/modules/miscellaneous/services/guichet-unique/forme.exercice.activite.principal.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { FormeExerciceActivitePrincipal } from 'src/app/modules/quotation/model/guichet-unique/referentials/FormeExerciceActivitePrincipal';
import { AppRestService } from 'src/app/services/appRest.service';

@Injectable({
  providedIn: 'root'
})
export class FormeExerciceActivitePrincipalService extends AppRestService<FormeExerciceActivitePrincipal>{

  constructor(http: HttpClient) {
    super(http, 'quotation/guichet-unique');
  }

  getFormeExerciceActivitePrincipal() {
    return this.getList(new HttpParams(), 'forme-exercice-activite-principal');
  }

}                        

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/guichetUnique/referentials/FormeExerciceActivitePrincipalService.java
package com.jss.osiris.modules.osiris.quotation.service.guichetUnique.referentials;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.guichetUnique.referentials.FormeExerciceActivitePrincipal;

public interface FormeExerciceActivitePrincipalService {
    public List<FormeExerciceActivitePrincipal> getFormeExerciceActivitePrincipal();

    public FormeExerciceActivitePrincipal getFormeExerciceActivitePrincipal(String code);
}
